### ISSUE TITLE
[v3] Media previews not working in v3 (using assets)

### ISSUE THREAD

**Issue by mediabeastnz** [sanity-io/sanity#3860]
**Describe the bug**

Media previews not working in V3 (using assets). Following the documentation and still no image appears in the preview only a blank spot.

Also tried existing V2 code but no cigar.

**To Reproduce**

Create the following type and check if a image preview shows up.

**Expected behavior**

A preview of the asset/media appears.

**Screenshots**

**Which versions of Sanity are you using?**

3.0.0 RC-2

**Which versions of Node.js / npm are you running?**

v16.14.2 / 8.5.0

**Additional context**

```ts
export default defineType({
  name: 'figure',
  title: 'Figure',
  type: 'image',
  options: {
    hotspot: true,
  },
  fields: [
    {
      title: 'Alternative text',
      name: 'alt',
      type: 'string',
      description: 'Important for SEO and accessiblity.',
      options: {
        isHighlighted: true
      },
      validation: Rule => {
        return Rule.custom((field, context) =>
          'asset' in context.parent && field === undefined
            ? 'Required! (think about non-visual readers)'
            : true
        )
      }
    }
  ],
  preview: {
    select: {
      alt: 'alt',
      media: 'asset',
    },
    prepare({ alt, media }) {
      return {
        title: alt,
        media: media
      };
    },
  },
```

**Comment by lasseaeggen** [comment]
As a temporary solution you can use the method described in discussion #3769

**Comment by bjoerge** [comment]
This should be fixed since `v3.0.0`. Feel free to open a new issue if that's not the case.
